File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.js
function solution(myString, pat) {
 let str = ''

  for (let i = 0; i < myString.length; i++) {
    if (myString[i] == "A") {
      str += "B"
    } else {
      str += "A"
    }
  }
  return str.indexOf(pat) +1 ? 1 : 0
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
class Solution {
  public int solution(String myString, String pat) {
    StringBuilder sb = new StringBuilder();
    
    for (String str : myString.split("")) {
      if (str.equals("A")) sb.append("B");
      else sb.append("A");
    }
    
    if (sb.toString().contains(pat)) return 1;
    else return 0;
  }
}